mpmp/utilities/tcga_utilities.py

```diff
@@ -321,34 +321,40 @@ def subsample_to_smallest_cancer_type(X_df,
     return X_ss_df, y_ss_df
 
 
+def get_overlap_data_types(use_subsampled=False, compressed_data=False):
+    """Get data types to restrict training samples to."""
+    if use_subsampled:
+        data_types = cfg.subsampled_data_types
+    elif compressed_data:
+        data_types = cfg.compressed_data_types
+    else:
+        data_types = cfg.data_types
+    return data_types
+
+
 def filter_to_cross_data_samples(X_df,
                                  y_df,
-                                 debug=False,
+                                 use_subsampled=False,
                                  verbose=False,
-                                 compressed_data=False,
+                                 compressed_data_only=False,
                                  n_dim=None):
     """Filter dataset to samples included in all data modalities."""
 
     # first, get intersection of samples in all training datasets
 
-    if debug:
-        data_types = cfg.subsampled_data_types
-    elif compressed_data:
-        data_types = cfg.compressed_data_types
-    else:
-        data_types = cfg.data_types
+    data_types = get_overlap_data_types(use_subsampled, compressed_data_only)
 
+    # TODO this may take some time to load, so we could cache it somewhere
```

Comment:
Unclear what 'this' is referring to, but if this is a function you call frequently you could use something like [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache) to do caching pretty easily